Examples/src/main/kotlin/com/mapbox/vision/examples/POIActivityKt.kt

```diff
@@ -0,0 +1,263 @@
+                val worldCoordinate = VisionReplayManager.geoToWorld(geoCoordinate) ?: return@mapNotNull null
+                val distanceToVehicle = latLng.distanceTo(currentVehicleLatLng).toInt()
+                POIState(it, distanceToVehicle, worldCoordinate)
+            }
+        }
+
+        // Show only POI which is close enough and behind the car
+        private fun filterPOIByDistance(poiStateList: List<POIState>) = poiStateList.filter {
+            val x = it.worldCoordinate.x
+            // Check if POI is behind vehicle and close enough to start appearing
+            (x > 0) && (it.distanceToVehicle < MIN_DISTANCE_METERS_FOR_DRAW_LABEL)
+        }
+
+        private fun preparePOIDrawData(poiStateList: List<POIState>): List<POIDrawData> = poiStateList.map { poiState ->
+            // Prepare bounding rect for POI in mobile screen coordinates
+            val poiBitmapRect = calculatePOIScreenRect(poiState.worldCoordinate)
+            val poiLabelAlpha = calculatePOILabelAlpha(poiState)
+            POIDrawData(poiState.poi.bitmap, poiBitmapRect, poiLabelAlpha)
+        }
+
+        private fun calculatePOIScreenRect(poiWorldCoordinate: WorldCoordinate): Rect {
+            // Calculate left top coordinate of POI in real world using POI world coordinate
+            val worldLeftTop = poiWorldCoordinate.copy(
+                    y = poiWorldCoordinate.y + LABEL_SIZE_METERS / 2,
+                    z = poiWorldCoordinate.z + LABEL_ABOVE_GROUND_METERS + LABEL_SIZE_METERS
+            )
+
+            // Calculate right bottom coordinate of POI in real world using POI world coordinate
+            val worldRightBottom = poiWorldCoordinate.copy(
+                    y = poiWorldCoordinate.y - LABEL_SIZE_METERS / 2,
+                    z = poiWorldCoordinate.z + LABEL_ABOVE_GROUND_METERS
+            )
+            val poiBitmapRect = Rect(0, 0, 0, 0)
+
+            // Calculate POI left top position on camera frame from real word coordinates
+            VisionReplayManager.worldToPixel(worldLeftTop)?.run {
+                poiBitmapRect.left = x
+                poiBitmapRect.top = y
+            }
+
+            // Calculate POI right bottom position on camera frame from real word coordinates
+            VisionReplayManager.worldToPixel(worldRightBottom)?.run {
+                poiBitmapRect.right = x
+                poiBitmapRect.bottom = y
+            }
+            return poiBitmapRect
+        }
+
+        private fun calculatePOILabelAlpha(poiState: POIState): Int {
+            val minDistance = min(MIN_DISTANCE_METERS_FOR_DRAW_LABEL - poiState.distanceToVehicle, DISTANCE_FOR_ALPHA_APPEAR_METERS)
+            return ((minDistance / DISTANCE_FOR_ALPHA_APPEAR_METERS.toFloat()) * 255).toInt()
+        }
+
+        private fun updateBitmapByPOIList(bitmap: Bitmap, poiDrawDataList: List<POIDrawData>) {
+            val canvas = Canvas(bitmap)
```

Comment:
I guess we could get rid of `bitmapCameraFrame` and hold `Canvas` instance instead - just set it in `onCameraUpdated`

Reply:
Bitmap is required for ImageView. But we can avoid redundant recreation of Canvas which is called quite frequently